elasticsearch.go

```diff
@@ -19,8 +20,16 @@ const (
 // Config represents the client configuration.
 //
 type Config struct {
-	Addresses []string          // A list of Elasticsearch nodes to use.
-	Transport http.RoundTripper // The HTTP transport object.
+	// A list of Elasticsearch nodes to use.
+	Addresses []string
+
+	// The HTTP transport object.
+	Transport http.RoundTripper
+
+	LogOutput       io.Writer             // Logging output
```

Comment:
As neat as this is, I think it would be better to have a Logger interface, and allow the user to provide an implementation, like:

```go
logger := &elasticsearch.TextLogger{Output: os.Stdout}
// or &elasticsearch.JSONLogger{Output: os.Stdout, LogResponseBody: true}
// or &elasticsearch.CurlLogger{Output: os.Stdout, LogRequestBody: true}
client, err := elasticsearch.NewClient(elasticsearch.Config{
    Logger: logger,
})
```

That would make it more straightforward to integrate into popular loggers like logrus, zap, zerolog, etc. While it would be possible to do the same with a custom RoundTripper, doing it this way would take care of copying the body and such.